fix(deps): update plugin-functions to 1.11.1

joshwlewis opened the pull request:
### What does this PR do?

This PR updates `plugin-functions` to the latest release. It's pretty far out of date at present, so the out of the box experience for functions isn't ideal.

### Acceptance Criteria

- Run `sf generate function -n myfunction -l javascript`
- Check that `cat functions/myfunction/project.toml` includes `salesforce-api-version = "54.0"`. Prior to this PR, this command generates a function with `salesforce-api-version = "53.0"`.

### What issues does this PR fix or reference?

- https://github.com/heroku/builder/pull/263#issuecomment-1154851050
- Supersedes #239 


### Other thoughts

Shouldn't dependabot have done this for us?

joshwlewis commented:
It looks like there have been some changes to commands that this has brought in:

- `env:log` @zwhitfield3 added this in https://github.com/salesforcecli/plugin-functions/pull/388
- `setalias`: @zwhitfield3 removed this in https://github.com/salesforcecli/plugin-functions/pull/300

@mimen @zwhitfield3 do y'all know if these went through the CLI review board process?

RodEsp reviewed (changes requested):
`env log` command needs to be marked as beta before this can be merged

joshwlewis commented:
> plugin-functions still needs to be bumped to the version that has the env log command marked as beta.

Updated to the latest `plugin-functions` and rebased. 🤞🏻

joshwlewis commented:
Updated to the latest plugin-functions (to 1.13.1) and rebased (again).
